Documentation/Guides/Error Recovery.md

```diff
@@ -1,3 +1,33 @@
+# Error Recovery
+
+When it comes to networking, errors are a given. It is very important that a networking framework give the client total control on determining which errors are recoverable and which are fatal, and a path from which to recover from non-fatal errors. Hyperspace provides two methods to recover from errors in the network transport process, described below.  
+
+## Quick Recovery
+
+Sometimes it is necessary to consider a status code that typically indicates a failure as a success. For example, some auth implementations return a redirect and ask the user to then manually extract information from the headers of that redirect response. In in this case, it is very easy to tell Hyperspace to consider a 3XX response a success:
+
+```swift
+func handlingRedirectsAsSuccess() -> Request<Response, Error> {
+       var copy = self
+       copy.recoveryTransformer = { failure in
+           guard let response = failure.response, response.status.isRedirection else { return nil }
+           return TransportSuccess(response: response)
+       }
+
+       return copy
+   }
+```
+
+```swift
+let request = Request<HTTP.Response, MyError> = .init(method: .get, url: url, headers: [.accept: .applicationJSON]) { transportSuccess in
+           return .success(transportSuccess.response)
+       }.handlingRedirectsAsSuccess()
+```
+
+Instead of going through the `TransportFailureRepresentable` machinery as usual, the present of a `Request.recoveryTransfer` that returns a non-nil `TransportSuccess` object, allows the request to be considered successful, giving the `Request.successTransformer` access to the raw 3XX `HTTP.Response`.
```

Comment:
How does this read?
```suggestion
Instead of going through the `TransportFailureRepresentable` machinery as usual, the presence of a `Request.recoveryTransfer` that returns a non-nil `TransportSuccess` object allows the request to be considered successful giving the `Request.successTransformer` access to the raw 3XX `HTTP.Response`.
```